extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResource.scala

```diff
@@ -60,6 +64,17 @@ object AccessResource {
       case TABLE | VIEW | INDEX =>
         resource.setValue("database", firstLevelResource)
         resource.setValue("table", secondLevelResource)
+      case URI =>
+        val objectList = new util.ArrayList[String]
+        if (firstLevelResource != null && firstLevelResource.nonEmpty) {
+          objectList.add(firstLevelResource)
+          if (!(firstLevelResource.substring(firstLevelResource.length - 1) == File.separator)) {
+            objectList.add(firstLevelResource + File.separator)
+          } else {
+            objectList.add(firstLevelResource.substring(firstLevelResource.length - 2))
+          }
+        }
```

Comment:
```scala
Option(firstLevelResource)
  .filter(_.nonEmpty)
  .map { path =>
     val s = path.stripSuffix(File.separator)
     Seq(s, s + File.separator)
   }.orNull
```

Reply:
Done

Reply:
@yaooqinn Why do we need process both `s` and `s + File.separator`?

Reply:
> @yaooqinn Why do we need process both `s` and `s + File.separator`?

Follow hive's code..